### PR TITLE
`rustc_resolve` cleanups

### PR DESCRIPTION
**Pull request by nnethercote**
Some improvements I found while looking through this code.

r? @estebank

### CODE REVIEW

File: compiler/rustc_resolve/src/diagnostics.rs [hunk 1]
```diff
@@ -3026,14 +3024,3 @@ fn is_span_suitable_for_use_injection(s: Span) -> bool {
     // import or other generated ones
     !s.from_expansion()
 }
-
-/// Convert the given number into the corresponding ordinal
-pub(crate) fn ordinalize(v: usize) -> String {
```
Comment (Nilstrieb) [inline compiler/rustc_resolve/src/diagnostics.rs:3031]: this conflicts with #125042, but I think that one can use just the number too

Comment (long-long-float) [reply compiler/rustc_resolve/src/diagnostics.rs:3031]: In https://github.com/rust-lang/rust/pull/125042, the function is copied, not referenced (put aside whether it is good or not).

File: compiler/rustc_resolve/src/effective_visibilities.rs [hunk 2]
```diff
@@ -11,6 +11,7 @@ use rustc_middle::middle::privacy::Level;
 use rustc_middle::middle::privacy::{EffectiveVisibilities, EffectiveVisibility};
 use rustc_middle::ty::Visibility;
 use std::mem;
+use tracing::info;
```
Comment (estebank) [inline compiler/rustc_resolve/src/effective_visibilities.rs:14]: Are all of these additions from tracing in different files needed?

**Review by estebank — APPROVED** [review]
@bors r+

File: tests/ui/lint/unused/unused-macro-rules.stderr [hunk 3]
```diff
@@ -10,13 +10,13 @@ note: the lint level is defined here
 LL | #![deny(unused_macro_rules)]
    |         ^^^^^^^^^^^^^^^^^^
 
-error: 2nd rule of macro `num` is never used
+error: rule #2 of macro `num` is never used
   --> $DIR/unused-macro-rules.rs:8:5
    |
 LL |     (two) => { 2 };
    |     ^^^^^
```
Comment (estebank) [inline tests/ui/lint/unused/unused-macro-rules.stderr:17]: Future note: we should emit a single error for all of the unused rules in a single macro.

File: compiler/rustc_resolve/src/effective_visibilities.rs (hunk 2 above)
Comment (nnethercote) [reply compiler/rustc_resolve/src/effective_visibilities.rs:14]: Yes. the `#[macro_use] extern crate tracing;` gave us implicit import of all `tracing` macros. With that gone, we must do explicit imports, and `tracing` macros are used in lots of places.

**Pull request merged**